aa_stripe/models.py

```diff
@@ -462,17 +469,17 @@ class StripeWebhook(models.Model):
 
     def _parse_coupon_notification(self, action):
         coupon_id = self.raw_data["data"]["object"]["id"]
+        created = timestamp_to_timezone_aware_date(self.raw_data["data"]["object"]["created"])
         if action == "created":
-            try:
-                StripeCoupon(coupon_id=coupon_id).save(force_retrieve=True)
-            except stripe.error.InvalidRequestError:
-                # do not fail in case the coupon has already been removed from Stripe before we received the webhook
-                pass
+            if StripeCoupon.objects.filter(coupon_id=coupon_id, created=created).exists():
+                raise StripeCouponAlreadyExists
+
+            StripeCoupon(coupon_id=coupon_id).save(force_retrieve=True)
         elif action == "updated":
-            StripeCoupon.objects.filter(coupon_id=coupon_id, is_deleted=False).update(
+            StripeCoupon.objects.filter(coupon_id=coupon_id, created=created, is_deleted=False).update(
                 metadata=self.raw_data["data"]["object"]["metadata"])
         elif action == "deleted":
-            StripeCoupon.objects.filter(coupon_id=coupon_id).update(is_deleted=True)
+            StripeCoupon.objects.filter(coupon_id=coupon_id, created=created).update(is_deleted=True)
```

Comment:
Same as above - post_save or any signal will not be triggered. Is this intentional? If so, why?

Reply:
Yes, we do not need to call api to delete the coupon (which won't change anything, the coupon is already deleted and does not exist at Stripe API).

Reply:
Delete should not trigger API call but post_save would be nice, if the system is listening for it.
IE: system could send an email to admin that stripe coupon of certain code was removed from stripe (failsafe),

Reply:
Actually, on second thought, it would be nice to use plain simple delete() call.

Reply:
Added queryset `delete()` which calls `StripeCoupon.save()`, so the post_save will be called.